lib/localhost/authority.rb

```diff
@@ -14,7 +14,7 @@ module Localhost
 	# Represents a single public/private key pair for a given hostname.
 	class Authority
 		def self.path
-			File.expand_path("~/.localhost")
+			File.expand_path("localhost.rb", ENV.fetch("XDG_STATE_HOME", "~/.local/state"))
```

Comment:
```suggestion
			state_home = ENV.fetch("XDG_STATE_HOME") do
				FileUtils.mkdir_p(DEFAULT_XDG_STATE_HOME)
			end
			
			return File.expand_path("localhost.rb", state_home)
```

Maybe something like this?

Reply:
(need to add `DEFAULT_XDG_STATE_HOME` + documentation)

The implication of this, is we ONLY create the default state directory IF the user hasn't specified it otherwise.

Reply:
So assuming that if the user has specified $XDG_STATE_HOME they are smart enough to ensure the directory exists. Seems reasonable.

Reply:
Also I don't know what permissions `FileUtils.mkdir_p` will use - might need to check and specify 0700

Reply:
> Also I don't know what permissions `FileUtils.mkdir_p` will use - might need to check and specify 0700

Good call - when mode is not specified it defaults to whatever `Dir.mkdir` uses, which is `0755`.

https://github.com/ruby/ruby/blob/master/lib/fileutils.rb#L399

Reply:
Ended up going with always creating the directory, rather than trying to be smarter.